ceramic-kubo-rpc/src/dag.rs

```diff
@@ -0,0 +1,85 @@
+//! Implements the dag endpoints.
+use std::io::{Cursor, Read, Seek};
+
+use anyhow::anyhow;
+use iroh_api::{Cid, IpfsPath};
+use libipld::{
+    cbor::DagCborCodec,
+    multihash::{Code, MultihashDigest},
+    pb::DagPbCodec,
+    prelude::{Codec, Decode, Encode},
+    Ipld,
+};
+
+use crate::{error::Error, IpfsDep};
+
+/// Get a DAG node from IPFS.
+#[tracing::instrument(skip(client, output_codec))]
+pub async fn get<T, C>(client: T, cid: Cid, output_codec: C) -> Result<Vec<u8>, Error>
```

Comment:
Kind of seems like all of these should be hidden behind a trait, so we're not passing client around.

Reply:
Can you expand on this a bit? The `client` is a trait already and these functions provide the crate level API we want to expose. What code structure are you expecting?